File: server/routes/chatbot.js
// import express from 'express';
// const router = express.Router();
// import Configuration from "openai";
// import OpenAIApi from "openai";
// import dotenv from 'dotenv'
// dotenv.config()

// const configuration = new Configuration({
//   apiKey: process.env.OPENAI_API_KEY,
// });
// console.log(configuration);
// const openai = new OpenAIApi(configuration);

// router.post('/ask', async (req, res) => {
//   const { topic, difficulty } = req.body;
//     console.log(topic);
//   if (!topic || !difficulty) {
//     return res.status(400).json({ error: 'Topic and difficulty are required' });
//   }

//   try {
//     const response = await openai.chat.completions.create({
//       model: 'gpt-3.5-turbo',
//       messages: [
//         { role: 'system', content: 'You are a chatbot that suggests project ideas.' },
//         { role: 'user', content: `Suggest a ${difficulty} level project idea on the topic: ${topic}.` },
//       ],
//     });

//     const suggestion = response.data.choices[0].message.content;
//     res.json({ suggestion });
//   } catch (error) {
//     console.error('Error in Chatbot Route:', error.message);
//     res.status(500).json({ error: 'Internal Server Error' });
//   }
// });


// export {router as ChatbotRouter}

File: python-backend/app/routes/chatbot.py
import os
from fastapi import APIRouter, HTTPException
from pydantic import BaseModel
from fireworks.client import Fireworks
from dotenv import load_dotenv

# Load environment variables
load_dotenv()

# API Router
router = APIRouter()

# Fireworks API Key
API_KEY = os.getenv("FIREWORKS_API_KEY")

if not API_KEY:
    raise ValueError("FIREWORKS_API_KEY is not configured in the environment variables")

# Initialize Fireworks client
client = Fireworks(api_key=API_KEY)

# Request model
class ChatbotRequest(BaseModel):
    topic: str
    difficulty: str

@router.post("/ask/")
async def ask_chatbot(request: ChatbotRequest):
    # Create prompt based on user input
    prompt = f"Suggest a {request.difficulty} project idea on {request.topic}.In the form of title,features description in which you will say what should it contain"
    
    try:
        # Generate response using Fireworks API
        response = client.chat.completions.create(
            model="accounts/fireworks/models/llama-v3-8b-instruct",
            messages=[
                {
                    "role": "user",
                    "content": prompt,
                }
            ],
        )
        
        # Extract the generated message
        suggestion = response.choices[0].message.content
        return {"suggestion": suggestion}
    
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"Error with Fireworks API: {str(e)}")
